### PR TITLE
Fixing Set-DbaAgentJob not being able to set some of the values to 0

### LINKED ISSUE #4927: Set-DbaAgentJob: Changing EventLogLevel to 0/Never will not be applied
### Before submitting a bug report:

- [x] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [x] Verified this bug is not already reported in an issue
- [x] Verified errors are not related to permissions
- [x] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

## Steps to Reproduce
When changing the EventLogLevel of a SQL Server Agent Job to 0/Never the modification will not be changed. The previous setting is retained. There is no error thrown or exeption using -EnableException. Changing the EventLogLevel to 1, 2 or 3 works as expected.

```powershell
$instance = 'Server\Instance'
Set-DbaAgentJob -SqlInstance $instance -Job 'Test1' -EventLogLevel 0 -Verbose -EnableException
VERBOSE: Performing the operation "Changing the job Test1" on target "DEFR2APP912\SCOMRS".
VERBOSE: [16:30:20][Set-DbaAgentJob] Changing the job

ComputerName           : Server
InstanceName           : Instance
SqlInstance            : Server\Instance
Name                   : Test1
Category               : [Uncategorized (Local)]
OwnerLoginName         : lab\adm
CurrentRunStatus       : Idle
CurrentRunRetryAttempt : 0
Enabled                : True
LastRunDate            : 01.01.0001 00:00:00
LastRunOutcome         : Unknown
HasSchedule            : False
OperatorToEmail        : 
CreateDate             : 09.01.2019 16:04:54

VERBOSE: [16:12:20][Set-DbaAgentJob] Finished changing job(s)



Get-DbaAgentJob -SqlInstance $instance | where Name -eq 'Test1' | select Eventloglevel,name

EventLogLevel Name 
------------- ---- 
    OnFailure Test1
```

When changing the EventLogLevel directly like the function does it works and the checkbox will be unchecked.
```powershell
$job = $server.JobServer.jobs | where Name -eq 'Test1'
$job.EventLogLevel = '0'
$job.Alter()

PS C:\>     $job.EventLogLevel
Never
```

## Expected Behavior
The EventLogLevel (Write to the Windows Application event log) checkbox schould be unchecked.
eventloglevel_expected

## Actual Behavior
The EventLogLevel won't be changed
eventloglevel_actual

## Environmental data
- dbatools:
Version 0.9.737

 - PowerShell:
PSVersion                      5.1.14393.2636                                                                                                                                                                                
PSEdition                      Desktop                                                                                                                                                                                       
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                       
BuildVersion                   10.0.14393.2636                                                                                                                                                                               
CLRVersion                     4.0.30319.42000                                                                                                                                                                               
WSManStackVersion              3.0                                                                                                                                                                                           
PSRemotingProtocolVersion      2.3                                                                                                                                                                                           
SerializationVersion           1.1.0.1

 - SQL Server:
Microsoft SQL Server 2008 R2 (SP3-GDR) (KB4057113) - 10.50.6560.0 (X64)   Dec 28 2017 15:03:48   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows NT 6.1 <X64> (Build 7601: Service Pack 1) 
Microsoft SQL Server 2012 (SP4-GDR) (KB4057116) - 11.0.7462.6 (X64)   Jan  5 2018 22:11:56   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor) 
Microsoft SQL Server 2014 (SP2-CU15) (KB4469137) - 12.0.5605.1 (X64)   Nov 30 2018 02:08:03   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor) 
Microsoft SQL Server 2016 (SP2-GDR) (KB4293802) - 13.0.5081.1 (X64)   Jul 20 2018 22:12:40   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor) 
Microsoft SQL Server 2016 (SP2-CU4) (KB4464106) - 13.0.5233.0 (X64)   Nov  3 2018 00:01:54   Copyright (c) Microsoft Corporation  Standard Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: )

### PR DESCRIPTION
**Pull request by nvarscar**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #4927)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Allow setting some values to 0

### Approach
`if($null -ne $value)` instead of `if ($value)`

### CODE REVIEW

File: functions/Set-DbaAgentJob.ps1 [hunk 1]
```diff
@@ -379,12 +379,12 @@ function Set-DbaAgentJob {
                 }
             }
 
-            if ($EventLogLevel) {
+            if ($null -ne $EventLogLevel) {
```
Comment (wsmelton) [inline functions/Set-DbaAgentJob.ps1]: This should use the `Test-Bound` internal function...

```suggestion
            if (Test-Bound -ParameterName EventLogLevel) {
```

File: functions/Set-DbaAgentJob.ps1 [hunk 2]
```diff
@@ -379,12 +379,12 @@ function Set-DbaAgentJob {
...
                 Write-Message -Message "Setting job event log level to $EventlogLevel" -Level Verbose
                 $currentjob.EventLogLevel = $EventLogLevel
             }
 
-            if ($EmailLevel) {
+            if ($null -ne $EmailLevel) {
```
Comment (wsmelton) [inline functions/Set-DbaAgentJob.ps1]:
```suggestion
            if (Test-Bound -ParameterName EmailLevel) {
```

File: functions/Set-DbaAgentJob.ps1 [hunk 3]
```diff
@@ -403,7 +403,7 @@ function Set-DbaAgentJob {
                 }
             }
 
-            if ($NetsendLevel) {
+            if ($null -ne $NetsendLevel) {
```
Comment (wsmelton) [inline functions/Set-DbaAgentJob.ps1]:
```suggestion
            if (if (Test-Bound -ParameterName NetsendLevel) {
```

File: functions/Set-DbaAgentJob.ps1 [hunk 4]
```diff
@@ -422,7 +422,7 @@ function Set-DbaAgentJob {
                 }
             }
 
-            if ($PageLevel) {
+            if ($null -ne $PageLevel) {
```
Comment (wsmelton) [inline functions/Set-DbaAgentJob.ps1]:
```suggestion
            if (if (Test-Bound -ParameterName PageLevel) {
```

File: functions/Set-DbaAgentJob.ps1 [hunk 5]
```diff
@@ -472,7 +472,7 @@ function Set-DbaAgentJob {
                 }
             }
 
-            if ($DeleteLevel) {
+            if ($null -ne $DeleteLevel) {
```
Comment (wsmelton) [inline functions/Set-DbaAgentJob.ps1]:
```suggestion
            if (if (Test-Bound -ParameterName DeleteLevel) {
```

File: functions/Set-DbaAgentJob.ps1 (hunk 1 above)
Comment (nvarscar) [reply functions/Set-DbaAgentJob.ps1]: That's not the same though. Using `Test-Bound` is to know when the parameter has been explicitly specified when the function was called. What would happen if a function (for whatever reason) was called like this:
```
Set-DbaAgentJob ... -EventLogLevel $null
```
It's definitely gonna throw an error, as the value is missing. Which is why I decided to go with `$null -ne ...`

Comment (wsmelton) [reply functions/Set-DbaAgentJob.ps1]: If that is the case, there are 491 references in the module that should be updated 🎐

Comment (wsmelton) [reply functions/Set-DbaAgentJob.ps1]: I would recommend simply updating `Test-Bound` to validate for null as well. Our standard for validating whether a parameter was passed in as been to use this internal function (we just have not been enforcing it on all the older commands).

Comment (nvarscar) [reply functions/Set-DbaAgentJob.ps1]: I checked and in this particular scenario the value cannot be null, as enforced by ValidateSet. Will change to Test-Bound

**Comment by potatoqualitee** [comment]
excellent, thank you!

**Pull request merged**